File: Stack/22GenerateParentheses/one.py
class Solution(object):
    def generateParenthesis(self, n):
        """
        :type n: int
        :rtype: List[str]
        """
        stack = []
        ans = []
        open = 0
        close = 0

        def backtrack(open,close):
            if(open == close == n):
                ans.append("".join(stack))
                return
            
            if(open<n):
                stack.append("(")
                backtrack(open+1,close)
                stack.pop()

            if(close<open):
                stack.append(")")
                backtrack(open,close+1)
                stack.pop()

        backtrack(open,close)
        return ans

    def main(self):
        n = 3
        ans = self.generateParenthesis(n)
        print(ans)

myInstance = Solution.main

File: BackTracking/386LexicographicalNumbers/one.js
/**
 * @param {number} n
 * @return {number[]}
 */
var lexicalOrder = function(n) {

    let ans = new Array()

    for(let i = 1; i<10; i++){
        backtrack(i,ans,n)

    }

    function backtrack(i,ans,n){
        
        if(i>n)
            return
        ans.push(i)
        for(let j =0; j<10;j++){
            backtrack(i*10+j,ans,n)
        }
    }

    console.log(ans)

};

let n = 192

lexicalOrder(n)